### ISSUE TITLE
Return the difference between value objects

### ISSUE THREAD

**Issue by ifdattic** [Ormin/phpspec-ddd-project-extension#2]
Wondering if it would be possible to return the difference between value objects?

In my opinion it would help as you could know what is the difference between VOs as current message is not that useful:

```
 34  - it returns its id
      expected ( value ) [obj:Domain\User\Value\UserId], but got [obj:Domain\User\Value\UserId].
```

**Comment by Ormin** [comment]
Good point.

Thing is , it might be hard to print an useful value of it, assuming that not every value object is a trivial one ( the one-fielders :) )

I'll check it in free time, meanwhile you're more than welcome to do so too and create a PR if you'll find a solution :+1:

**Comment by ifdattic** [comment]
Yep, my thoughts are the same that it can become complex depending on the value object.

Still the way I imagine that if you got something like a diff between the objects it would be a wow factor :D

**Comment by Ormin** [comment]
Hmm, perhaps recursively scanning the object up until the primitives and comparing? Would be definitely a lovely feature :+1: !

**Comment by ifdattic** [comment]
Can't remember who did something similar the way I imagine it (maybe PHPUnit with arrays...). This would probably be a good feature to iterate over time to take care of complexity. The simplest way would probably be to convert objects to array and compare them, but as the variables are private it probably won't work well. Otherwise I think just going over object properties and creating an array from them should work as well. Guess it would easily take care of 80% of use cases.
